examples/pytorch/ogb/directional_GSN/main.py

```diff
@@ -0,0 +1,433 @@
+from ogb.graphproppred import Evaluator
+import torch
+import numpy as np
+from dgl.dataloading import GraphDataLoader
+from tqdm import tqdm
+import dgl
+import random
+import torch.nn as nn
+from ogb.graphproppred.mol_encoder import AtomEncoder
+import torch.nn.functional as F
+from functools import partial
+import torch.optim as optim
+import argparse
+from torch.utils.data import Dataset
+from dgl.data.utils import Subset
+from preprocessing import prepare_dataset
+
+def aggregate_mean(h, vector_field, h_in):
+    return torch.mean(h, dim=1)
+
+def aggregate_max(h, vector_field, h_in):
+    return torch.max(h, dim=1)[0]
+
+def aggregate_sum(h, vector_field, h_in):
+    return torch.sum(h, dim=1)
+
+def aggregate_dir_dx(h, vector_field, h_in, eig_idx):
+    eig_w = ((vector_field[:, :, eig_idx]) /
+             (torch.sum(torch.abs(vector_field[:, :, eig_idx]), keepdim=True, dim=1) + 1e-8)).unsqueeze(-1)
+    h_mod = torch.mul(h, eig_w)
+    return torch.abs(torch.sum(h_mod, dim=1) - torch.sum(eig_w, dim=1) * h_in)
+
+AGGREGATORS = {'mean': aggregate_mean, 'sum': aggregate_sum, 'max': aggregate_max,
+               'dir1-dx': partial(aggregate_dir_dx, eig_idx=1)}
+
+class MLP(nn.Module):
+    def __init__(self, in_size, hidden_size, out_size, 
+                 dropout=0., last_b_norm=False, device='cpu'):
+        super(MLP, self).__init__()
+
+        self.in_size = in_size
+        self.hidden_size = hidden_size
+        self.out_size = out_size
+
+        self.fully_connected = nn.ModuleList()
+        self.fully_connected.append(FCLayer(in_size, out_size, b_norm=last_b_norm,
+                                            device=device, dropout=dropout))
+
+    def forward(self, x):
+        for fc in self.fully_connected:
+            x = fc(x)
+        return x
+
+class FCLayer(nn.Module):
+    def __init__(self, in_size, out_size, device, dropout=0., b_norm=False, bias=True):
+        super(FCLayer, self).__init__()
+
+        self.__params = locals()
+        del self.__params['__class__']
+        del self.__params['self']
+        self.in_size = in_size
+        self.out_size = out_size
+        self.bias = bias
+        self.linear = nn.Linear(in_size, out_size, bias=bias).to(device)
+        self.dropout = None
+        self.b_norm = None
+        if dropout:
+            self.dropout = nn.Dropout(p=dropout, device=device)
+        if b_norm:
+            self.b_norm = nn.BatchNorm1d(out_size).to(device)
+        self.init_fn = nn.init.xavier_uniform_
+
+        self.reset_parameters()
+
+    def reset_parameters(self, init_fn=None):
```

Comment:
```suggestion
    def reset_parameters(self):
```
`init_fn` is redundant.